### PR TITLE
editorconfig-checker: 2.0.2 -> 2.0.3

### PR DESCRIPTION
**Pull request by Frostman**
###### Motivation for this change

Bump to the latest version [0] and additionally fix package build as it's failing currently with module hash mismatch:
```
hash mismatch in fixed-output derivation '/nix/store/6ghdnnzr7w8ah9p6pym8mik67bp580jy-editorconfig-checker-2.0.2-go-modules':
  wanted: sha256:09b1v9gyh6827yqlfxxxq3lcqhd5snn3n7gdlbjmga3wyp2x4g2r
  got:    sha256:1iiv12ginb3ky739z7v8wf4z5lv24gmghbybs3lzay0kqn449n4x
cannot build derivation '/nix/store/dhickjqkym3hgr7ispji7yi7argwbnpj-editorconfig-checker-2.0.2.drv': 1 dependencies couldn't be built
```

[0] https://github.com/editorconfig-checker/editorconfig-checker/releases/tag/2.0.3

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/nixos/nixpkgs/79145)

### CODE REVIEW

**Review by uri-canva — APPROVED** [review]

**Comment by uri-canva** [comment]
> fix package build as it's failing currently with module hash mismatch

That's strange, it has a `go.sum` file so the modules should be completely reproducible.

I don't have merge permissions so I can't merge this, but it looks good to me.

**Comment by Frostman** [comment]
@uri-canva yeah, that's quite strange, I have no idea how it could happen with go.sum in the repo...

**Review by jonringer — APPROVED** [review]
diff LGTM
executable shows usage

```
[4 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/79145
1 package built:
editorconfig-checker
```

**Comment by jonringer** [comment]
> > fix package build as it's failing currently with module hash mismatch
> 
> That's strange, it has a `go.sum` file so the modules should be completely reproducible.
> 
> I don't have merge permissions so I can't merge this, but it looks good to me.

Sometimes the way the git objects are packed can affected the sha (generally only an issue with fetchgit+deepClone). Also darwin has some text unicode normalization which can also affect the sha.

**Pull request merged**
